Assets/MRTK/SDK/Editor/Inspectors/UX/Constraints/ConstraintManagerInspector.cs

```diff
@@ -0,0 +1,367 @@
+        /// Util method for drawing a consistent constraints section.
+        /// Use this method in a component inspector for linking to a constraint manager. 
+        /// </summary>
+        /// <param name="gameObject">Game object the constraint manager is attached to.</param>
+        /// <param name="managerEnabled">Serialized property for enabling the manager - needs to be of type bool.</param>
+        /// <param name="managerRef">Serialized property of the constraint manager component link - needs to be type of ConstraintManager.</param>
+        /// <param name="isExpanded">Flag for indicating if the constraint foldout was previously collapsed or expanded.</param>
+        /// <returns>Current state of expanded or collapsed constraint foldout. Returns true if expanded / contents visible.</returns>
+        static public bool DrawConstraintManagerFoldout(GameObject gameObject, SerializedProperty managerEnabled, SerializedProperty managerRef, bool isExpanded)
+        {
+            isExpanded = EditorGUILayout.Foldout(isExpanded, "Constraints", true);
+
+            if (isExpanded)
+            {
+                EditorGUILayout.PropertyField(managerEnabled);
+                GUI.enabled = managerEnabled.boolValue;
+                // Make sure we're having at least one constraint manager available.
+                // Usually this should be ensured by the component requirement. However 
+                // for components that had this requirement added after they were serialized
+                // this won't work out of the box.
+                gameObject.EnsureComponent<ConstraintManager>(); 
+                var constraintManagers = gameObject.GetComponents<ConstraintManager>();
+
+                int selected = 0;
+
+                string[] options = new string[constraintManagers.Length];
+
+                int manualSelectionCount = 0;
+                for (int i = 0; i < constraintManagers.Length; ++i)
+                {
+                    var manager = constraintManagers[i];
+                    if (managerRef.objectReferenceValue == manager)
+                    {
+                        selected = i;
+                    }
+
+                    // popups will only show unqiue elements
+                    // in case of auto selection we don't care which one we're selecting as the behavior will be the same.
+                    // in case of manual selection users might want to differentiate which constraintmanager they are referring to.
+                    if (manager.AutoConstraintSelection == true)
+                    {
+                        options[i] = manager.GetType().Name + " (auto)";
+                    }
+                    else
+                    {
+                        manualSelectionCount++;
+                        options[i] = manager.GetType().Name + " (manual " + manualSelectionCount + ")";
+                    }
+                }
+
+
+                EditorGUILayout.BeginHorizontal();
+
```

Comment:
Might be personal choice, but opting for 
`using (new EditorGUILayout.HorizontalScope()) {}` and `using (var check = new EditorGUI.ChangeCheckScope()){}`
instead of   `EditorGUILayout.BeginHorizontal()` and `EditorGUI.BeginChangeCheck()` might improve readability, and avoid forgetting to End the scope, which sometimes does not throw error straight away.